src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +20,45 @@ public static Injector getInjector() {
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Object clazzImplementationInstance = createNewInstance(clazz);
+        for (Field field : clazz.getDeclaredFields()) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = createNewInstance(findImplementation(field.getType()));
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (clazz.isAnnotationPresent(Component.class)) {
+            try {
+                Constructor<?> constructor = clazz.getConstructor();
+                return constructor.newInstance();
+            } catch (NoSuchMethodException | InvocationTargetException
+                     | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException("Can`t create new instance of " + clazz.getName());
+            }
+        } else {
+            throw new RuntimeException("Can`t create an instance of the class."
+                    + "There is no annotation");
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> implementations = new HashMap<>();
+        implementations.put(ProductService.class, ProductServiceImpl.class);
+        implementations.put(ProductParser.class, ProductParserImpl.class);
+        implementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        if (interfaceClazz.isInterface()) {
+            return implementations.get(interfaceClazz);
+        }
+        return interfaceClazz;
```

Comment:
The implementations map should be a class field and initialized using Map.of() for better performance and readability.